ts/Data/Parsers/HTMLTableParser.ts

```diff
@@ -59,15 +76,30 @@ class HTMLTableParser extends DataParser<DataParser.EventObject> {
      *
      * */
 
+    /**
+     * Constructs an instance of the HTML table parser.
+     *
+     * @param {HTMLElement | null} tableElement
+     * The HTML table to parse
+     * @param {HTMLTableParser.OptionsType} [options]
+     * Options for the CSV parser.
+     */
     constructor(
-        tableElement: (HTMLElement|null) = null,
-        options?: Partial<HTMLTableParser.Options>
+        tableElement: (HTMLElement | null) = null,
+        options?: Partial<HTMLTableParser.OptionsType>
     ) {
         super();
         this.columns = [];
         this.headers = [];
         this.options = merge(HTMLTableParser.defaultOptions, options);
-        this.tableElement = tableElement;
+
+        if (tableElement) {
+            this.tableElement = tableElement;
+            this.tableElementID = tableElement.id;
+        } else if (options?.tableHTML) {
+            this.tableElement = options.tableHTML;
+            this.tableElementID = options?.tableHTML.id;
+        }
     }
```

Comment:
Thoughts on the parser constructors? Should we try to stick with `constructor(element, options)` or just `constructor(options)`.

Currently the CSVParser constructor only deals with an options object. The HTMLTableParser does both the element to parse and options (with the element also being an option, which should probably change).

Reply:
It could be beneficial to share constructor arguments between parsers. The HTMLTableParser is probably the exception when it comes to the need of a DOM element. Maybe just switch the arguments order?